modules/linear_boltzmann_solvers/a_lbs_solver/distributed_source/distributed_source.cc

```diff
@@ -0,0 +1,82 @@
+#include "distributed_source.h"
+
+#include "modules/linear_boltzmann_solvers/a_lbs_solver/lbs_solver.h"
+
+#include "framework/mesh/cell/cell.h"
+#include "framework/mesh/mesh_continuum/mesh_continuum.h"
+#include "framework/mesh/logical_volume/logical_volume.h"
+#include "framework/math/functions/spatial_material_function.h"
+
+#include "framework/runtime.h"
+#include "framework/object_factory.h"
+
+namespace opensn::lbs
+{
+
+OpenSnRegisterObject(lbs, DistributedSource);
+
+InputParameters
+DistributedSource::GetInputParameters()
+{
+  InputParameters params = Object::GetInputParameters();
+
+  params.SetGeneralDescription("General implementation of an arbitrary multi-group "
+                               "distributed source. A distributed source is given by a "
+                               "logical volume and a Lua function handle. Currently, only "
+                               "isotropic distributed sources are allowed.");
+  params.SetClassName("Distributed Source");
+  params.SetDocGroup("LBSUtilities");
+
+  params.AddRequiredParameter<size_t>("logical_volume_handle",
+                                      "Handle to the logical volume the distributed source is "
+                                      "defined within.");
+  params.AddOptionalParameter(
+    "function_handle",
+    1000000,
```

Comment:
Seems like we are not able to specify an optional parameter without providing a default value. So, may be define something like `INVALID_LUA_HANDLE` and use that instead of a magic number.

Reply:
Define as in a `#define INVALID_LUA_HANDLE 1000000`? Where should this go?